examples/weak.rs

```diff
@@ -27,10 +27,10 @@ fn main() {
     storage.write().access_mut(&node2).next = Some(node1.downgrade());
 
     for node in storage.read().iter() {
-        let value: String = node.next.as_ref().map_or("None".into(), |ref next| {
-            storage.read()
-            .access_weak(next)
-            .ok().map_or("None".into(), |item| item.value.clone())
+        let value = node.next.as_ref().map_or("None".into(), |ref next| {
```

Comment:
this seems quite complicated, I'm going to think on reducing it, but it's not essential to the PR